pkg/cmd/update/update.go

```diff
@@ -0,0 +1,286 @@
+package update
+
+import (
+	"archive/tar"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/charmbracelet/glamour"
+	"github.com/coreeng/corectl/pkg/cmdutil/config"
+	"github.com/coreeng/corectl/pkg/cmdutil/shell"
+	"github.com/coreeng/corectl/pkg/cmdutil/userio"
+	"github.com/coreeng/corectl/pkg/cmdutil/userio/confirmation"
+	"github.com/coreeng/corectl/pkg/git"
+	"github.com/coreeng/corectl/pkg/version"
+	"github.com/google/go-github/v59/github"
+	"github.com/phuslu/log"
+	"github.com/spf13/cobra"
+)
+
+type UpdateOpts struct {
+	githubToken      string
+	streams          userio.IOStreams
+	targetVersion    string
+	skipConfirmation bool
+}
+
+// Any failures we recieve will log a warning, we don't want this to cause any command to fail, this is an optional
+// check which shouldn't prevent or interrupt any command from running (especially in ci)
+func CheckForUpdates(cfg *config.Config, cmd *cobra.Command) {
+	updateInterval := 1 * time.Hour
+	updateStatusFileName := "corectl-autoupdate"
+	log.Debug().Msg("checking for updates")
+	nonInteractive, err := cmd.Flags().GetBool("non-interactive")
```

Comment:
I would propose to also try to deduce this flag automatically by checking if we are working with a terminal or not.
Similar to what we do here: https://github.com/coreeng/corectl/blob/829d30c28592d75ac707ef496359e7450cd5fb87/pkg/cmdutil/userio/prompt.go#L65
This flag was meant to be just as an option to explicitly disable all interactivity